app/scripts/modules/core/src/task/TaskProgressBar.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+
+import { ITask } from 'core/domain';
+import { robotToHuman } from 'core/presentation/robotToHumanFilter/robotToHuman.filter';
+
+export interface ITaskProgressBarProps {
+  task: ITask;
+}
+
+export function TaskProgressBar(props: ITaskProgressBarProps) {
```

Comment:
I think we're using  either `class TaskProgressBar extends Component...` or `const TaskProgressBar = (props: ITaskProgressBarProps) => ` everywhere else (as opposed to `function`)

Reply:
I actually just followed what we were doing in other places. For example [ExecutionStepDetails](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/config/stages/core/ExecutionStepDetails.tsx#L13).

But we should probably pick one pattern to use moving forward.

Reply:
yeah IME we're very inconsistent. My personal tendency has been to just go with whatever is in the file already, or use an arrow function if brand new.